src/lib/engine/facts.js

```diff
@@ -94,33 +95,32 @@ export function addFactsToSpaceGraph({metrics, guesstimates, simulations}, globa
   // When dynamically generating new metrics, we need non-colliding readableIds, so we'll store a running copy of those
   // used, and initialize some variables to account for the extra objects we'll build.
   let readableIds = metrics.map(m => m.readableId)
-  let [factMetrics, factGuesstimates, factSimulations, factHandleMap] = [[], [], [], {}]
+  let [factMetrics, factGuesstimates, factSimulations, factIdMap] = [[], [], [], {}]
   grouped.forEach(([handle, selector, fact]) => {
     // We construct virtual metrics, guesstimates, and simulations from the selector, fact, and running readable IDs...
     const {metric, guesstimate, simulation} = buildFullNode(selector, fact, readableIds)
 
     // Then update all the running variables.
-    factHandleMap[handle] = metric.readableId
+    const idToTranslate = !!_.get(fact, 'id') ? `\$\{fact:${fact.id}\}` : handle
```

Comment:
I'm not using `_.has` intentionally here, because if the fact is, for example ```
{id: null, ...}
```
then `_.has` would return true, whereas I want it to return false.

Agreed about fact being constant, but this is a transient bit of code anyways, so I'm hesitant to spend much effort cleaning it up. Do you think it is worth it, as removing this is the next feature on my list regardless?